### PR TITLE
Add functionality for subscriber tier

### PR DESCRIPTION
**Pull request by offcast-dev**
PT: https://www.pivotaltracker.com/story/show/178026302
### User Story
```
As an API
To present Subscriber their movie of interest
I need to be able to search using query param
```
________
PT: https://www.pivotaltracker.com/story/show/178082568
### User Story
```
As an API
To provide access to premium content
We have to have functionality to take payments
```
________
### Changes Proposed
- Adds GET Search method using query param
- Add Stripe to get payments and upgrade User to Subscriber
- Allow Search method  to Subscriber tier only
### Things Learned
- How to set up Stripe for one-time payments on backend
- Change user status in tests

### CODE REVIEW

File: app/channels/application_cable/channel.rb [hunk 1]
```diff
@@ -1,4 +0,0 @@
-module ApplicationCable
```
Comment (DavDan1) [inline app/channels/application_cable/channel.rb:1]: why did you remove the files? don't you need this?

**Review by DavDan1 — APPROVED** [review]

File: app/channels/application_cable/channel.rb (hunk 1 above)
Comment (offcast-dev) [reply app/channels/application_cable/channel.rb:1]: We did this to increase test coverage, this files are not used in the application
